app/controllers/Reindex.scala

```diff
@@ -7,43 +7,51 @@ import play.api.libs.functional.syntax._
 import play.api.libs.json._
 import play.api.Logger
 import play.api.mvc.Controller
-
-case class ExpectedItemCount(expectedItemCount: Int)
-
-object ExpectedItemCount {
-  implicit val expectedItemCountFormat: Format[ExpectedItemCount] = (
-    JsPath \ "expectedItemCount"
-  ).format[Int].inmap(count => ExpectedItemCount(count), (itemsCount: ExpectedItemCount) => itemsCount.expectedItemCount)
-}
-
+import repositories.ReindexProgressRepository
 
 object Reindex extends Controller with PanDomainAuthActions {
   def reindexTags = (APIAuthAction andThen ReindexPermissionsCheck) { req =>
     // Get the reindex id provided by CAPI
-    req.body.asJson.map { json =>
-        try {
-          ReindexTagsCommand((json \ "jobId").as[String]).process.map{ count =>
-            Ok(Json.toJson(ExpectedItemCount(count)))
-          } getOrElse InternalServerError
-        } catch {
-          commandErrorAsResult
-        }
-      }.getOrElse {
-        BadRequest("Expecting reindex id in body")
+    try {
+      if (ReindexProgressRepository.isTagReindexInProgress) {
+        Forbidden
+      } else {
+        ReindexTagsCommand().process.map{ count =>
+          Ok
+        } getOrElse InternalServerError
       }
+    } catch {
+      commandErrorAsResult
+    }
   }
 
   def reindexSections = (APIAuthAction andThen ReindexPermissionsCheck) { req =>
-    req.body.asJson.map { json =>
-        try {
-          ReindexSectionsCommand((json \ "jobId").as[String]).process.map{ count =>
-            Ok(Json.toJson(ExpectedItemCount(count)))
-          } getOrElse InternalServerError
-        } catch {
-          commandErrorAsResult
-        }
-      }.getOrElse {
-        BadRequest("Expecting reindex id in body")
+    try {
+      if (ReindexProgressRepository.isSectionReindexInProgress) {
+        Forbidden
```

Comment:
429 would perhaps be more appropriate indeed. Apparently CAPI is expecting a 403 if there's on already running. I'd be more than happy to change it if CAPI can accept 429 or something more meaningful than forbidden.